app/models/user_mentee_application.rb

```diff
@@ -28,9 +29,50 @@
 #
 class UserMenteeApplication < ApplicationRecord
   belongs_to :user
+  has_many :mentee_application_states, dependent: :destroy
 
   validates :available_hours_per_week, numericality: { greater_than: 0, less_than_or_equal_to: 60 }
 
   validates :city, :state, :country, :reason_for_applying, :learned_to_code, :project_experience,
     :available_hours_per_week, presence: true
+
+  after_create :create_initial_application_state
+
+  delegate :rejected?, to: :current_state
+
+  delegate :accepted?, to: :current_state
+
+  def current_status
+    mentee_application_states.last&.status&.to_sym
```

Comment:
why are we using the safe navigation operator here? Shouldn't there always be an associated state with an application?

Reply:
The more I think about it, the more this method seems like presentational/view logic that is mostly duplicating the `current_state` logic. Can we consolidate?

Reply:
For your first question: yes, good point, I iterated on this method a couple of times but eventually fixed this issue outside  outside of this method. don't think i need it anymore.

for the second: hmm, i'm not sure I get the question. can you elaborate?  This method checks the status of an user_mentee_application record. I use for view logic and in other methods, the state returns the whole mentee_application_state record.

Reply:
so, for the second: `current_state` and `current_status` are roughly the same, so I think we should choose one. Given that it is querying the `status` of the `MenteeApplicationState`, I'd recommend `current_status`.

Re: the view logic aspect, I'm wary of adding methods that support view logic to models to combat model bloat. Instead, if there need to be transformations of data to a more human-readable format, I tend to reach for presenters, decorators, or view helpers to define those methods.

Does that make sense/seem reasonable?

Reply:
I think so, but why don't we pair on this? I have time tomorrow am? 9am pacific? or let me know what time works better for you